### ISSUE TITLE
java.lang.NullPointerException

### ISSUE THREAD

**Issue by hrach** [JanTvrdik/intellij-latte#26]
<details>
<summary>StackTrace</summary>
null
java.lang.NullPointerException
	at com.jantvrdik.intellij.latte.inspections.DeprecatedMacroInspection$1.visitElement(DeprecatedMacroInspection.java:41)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.jantvrdik.intellij.latte.psi.impl.LatteMacroTagImpl.accept(LatteMacroTagImpl.java:26)
	at com.jantvrdik.intellij.latte.psi.impl.LatteMacroOpenTagImpl.accept(LatteMacroOpenTagImpl.java:25)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:64)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:26)
	at com.intellij.util.WalkingState.walkChildren(WalkingState.java:65)
	at com.intellij.util.WalkingState.elementStarted(WalkingState.java:52)
	at com.intellij.psi.PsiWalkingState.elementStarted(PsiWalkingState.java:73)
	at com.intellij.psi.PsiRecursiveElementWalkingVisitor.visitElement(PsiRecursiveElementWalkingVisitor.java:48)
	at com.jantvrdik.intellij.latte.inspections.DeprecatedMacroInspection$1.visitElement(DeprecatedMacroInspection.java:48)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at com.jantvrdik.intellij.latte.psi.impl.LatteMacroClassicImpl.accept(LatteMacroClassicImpl.java:26)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:199)
	at com.intellij.psi.impl.source.PsiFileImpl.acceptChildren(PsiFileImpl.java:820)
	at com.jantvrdik.intellij.latte.inspections.DeprecatedMacroInspection.checkFile(DeprecatedMacroInspection.java:35)
	at com.intellij.codeInspection.LocalInspectionTool$1.visitFile(LocalInspectionTool.java:156)
	at com.intellij.extapi.psi.PsiFileBase.accept(PsiFileBase.java:70)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:69)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.a(LocalInspectionsPass.java:294)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.access$200(LocalInspectionsPass.java:73)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:261)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass$2.process(LocalInspectionsPass.java:256)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:128)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:217)
	at com.intellij.concurrency.ApplierCompleter.c(ApplierCompleter.java:146)
	at com.intellij.concurrency.ApplierCompleter.access$000(ApplierCompleter.java:44)
	at com.intellij.concurrency.ApplierCompleter$1.run(ApplierCompleter.java:91)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1178)
	at com.intellij.concurrency.ApplierCompleter$2.run(ApplierCompleter.java:100)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:443)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.concurrency.ApplierCompleter.a(ApplierCompleter.java:112)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:88)
	at jsr166e.CountedCompleter.exec(CountedCompleter.java:684)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)
</details>

**Comment by hrach** [comment]
possible template
```
{snippet grid}
{php
    $form->getElementPrototype()->class[] = 'grido';

    $operation = $control->hasOperation();
    $actions = $control->hasActions() ? $control->getComponent(\Grido\Components\Actions\Action::ID)->getComponents() : array();

    $filters = $control->hasFilters() ? $form->getComponent(\Grido\Components\Filters\Filter::ID)->getComponents() : array();
    $filterRenderType = $control->getFilterRenderType();

    $columns = $control->getComponent(\Grido\Components\Columns\Column::ID)->getComponents();
    $columnCount = count($columns) + ($operation ? 1 : 0);
    $showActionsColumn = $actions || ($filters && $filterRenderType == \Grido\Components\Filters\Filter::RENDER_INNER);

    /* TWITTER BOOTSTRAP 2/3 */
    $buttons = $form->getComponent('buttons');
    $buttons->getComponent('search')->getControlPrototype()->class[] = 'btn btn-default btn-xs btn-mini search';
    $buttons->getComponent('reset')->getControlPrototype()->class[] = 'btn btn-default btn-xs btn-mini reset';

    $form['count']->controlPrototype->class[] = 'form-control';
    $operation ? $form['operations']['operations']->controlPrototype->class[] = 'form-control' : NULL}

    {foreach $filters as $filter}
        {php $filter->controlPrototype->class[] = 'form-control'}
    {/foreach}

    {foreach $actions as $action}
        {php
            $element = $action->getElementPrototype();
            $element->class[] = 'btn btn-default btn-xs btn-mini';
        }
        {if $icon = $action->getOption('icon')}
        {php
            $element->setText(''); /* $element->setText(' ' . $action->getLabel()); */
            $element->insert(0, \Nette\Utils\Html::el('i')->setClass(array("glyphicon glyphicon-$icon fa fa-$icon icon-$icon")));
        }
        {/if}
    {/foreach}
{if $form->getErrors()}
<ul n:foreach="$form->getErrors() as $error">
    <li>{$error}</li>
</ul>
{/if}

{form form}
{block table}
<div n:class="grid-wrapper">
<table class="table table-bordered table-condensed table-hover">
    <thead>
        <tr class="head">
            <th n:if="$operation" class="checker"{if $filters} rowspan="{if $filterRenderType == \Grido\Components\Filters\Filter::RENDER_OUTER}1{else}2{/if}"{/if}>
                <input type="checkbox" title="{_'Grido.Invert'}">
            </th>
            <th n:if="$showActionsColumn" class="actions center">
                {_'Grido.Actions'}
            </th>
            {foreach $columns as $column}
                {php $column->getHeaderPrototype()->class[] = $column->isSortable() ? (sorting . ($column->getSort() ? ' sorting-' . $column->getSort()))}
                {$column->getHeaderPrototype()->startTag()|noescape}
                    {if $column->isSortable()}
                        <a n:if="!$column->getSort()" n:href="sort! [$column->getName() => \Grido\Components\Columns\Column::ORDER_ASC]">{$column->getLabel()|noescape}</a>
                        <a n:if="$column->getSort() == \Grido\Components\Columns\Column::ORDER_ASC" n:href="sort! [$column->getName() => \Grido\Components\Columns\Column::ORDER_DESC]" class="sort">{$column->getLabel()|noescape}</a>
                        <a n:if="$column->getSort() == \Grido\Components\Columns\Column::ORDER_DESC" n:href="sort! [$column->getName() => \Grido\Components\Columns\Column::ORDER_ASC]" class="sort">{$column->getLabel()|noescape}</a>
                        <span></span>
                    {else}
                        {$column->getLabel()|noescape}
                    {/if}
                {$column->getHeaderPrototype()->endTag()|noescape}
            {/foreach}
        </tr>
        <tr n:if="$filterRenderType == \Grido\Components\Filters\Filter::RENDER_INNER && $filters" class="filter inner">
            <th n:if="$filters" class="buttons" n:block="action" style="padding: 0 !important; vertical-align: middle;">
                {formContainer buttons}
                    {input search, style => 'margin-right: 3px'}{input reset}
                {/formContainer}
            </th>
            {foreach $columns as $column}
                {if $column->hasFilter()}
                    {php $control->getFilter($column->getName())->getWrapperPrototype()->class[] = 'filter'}
                    {$control->getFilter($column->getName())->getWrapperPrototype()->startTag()|noescape}
                    {formContainer filters}
                        {input $column->getName(), size => 1}
                    {/formContainer}
                    {$control->getFilter($column->getName())->getWrapperPrototype()->endTag()|noescape}
                {elseif $column->headerPrototype->rowspan != 2}
                    <th>&nbsp;</th>
                {/if}
            {/foreach}
        </tr>
    </thead>
    <tbody>
        {var $propertyAccessor = $control->getPropertyAccessor()}
        {foreach $data as $row}
            {php $checkbox = $operation
                    ? $form[\Grido\Components\Operation::ID][\Grido\Helpers::formatColumnName($propertyAccessor->getValue($row, $control->getComponent(\Grido\Components\Operation::ID)->getPrimaryKey()))]
                    : NULL;
                $tr = $control->getRowPrototype($row);
                $tr->class[] = $checkbox && $checkbox->getValue()
                    ? 'selected'
                    : NULL;
            }
            {$tr->startTag()|noescape}
                <td n:if="$checkbox" class="checker">
                    {$checkbox->getControl()}
                </td>
                <td n:if="$showActionsColumn" class="actions center">
                    {foreach $actions as $action}
                        {control $action $row}
                    {/foreach}
                    {if !$actions}
                        &nbsp;
                    {/if}
                </td>
                {foreach $columns as $column}
                    {php $td = $column->getCellPrototype($row)}
                    {$td->startTag()|noescape}
                        {if is_string($column->getCustomRender()) && $column->getCustomRenderVariables()}
                            {include $column->getCustomRender() control => $control, presenter => $control->getPresenter(), item => $row, column => $column, (expand) $column->getCustomRenderVariables()}
                        {elseif is_string($column->getCustomRender())}
                            {include $column->getCustomRender() control => $control, presenter => $control->getPresenter(), item => $row, column => $column}
                        {else}
                            {$column->render($row)|noescape}
                        {/if}
                    {$td->endTag()|noescape}
                {/foreach}
            {$tr->endTag()|noescape}
        {/foreach}
        <tr n:if="!$control->getCount()"><td colspan="{=$showActionsColumn ? $columnCount + 1 : $columnCount}" class="no-results">{_'Grido.NoResults'}</td></tr>
    </tbody>
</table>
</div>
{/block}

{block footer}
{var $form = $control->getComponent('form')}
{var $paginator = $control->getPaginator()}

<div id="footer">
<span n:block="operations" class="operations" title="{_'Grido.SelectSomeRow'}">
	{if $newButtonLink}
	<a class="btn btn-success pull-left" href="{$newButtonLink}">
		<span class="glyphicon glyphicon-plus"></span>
	</a>
	{/if}
	{if $control->hasOperation()}
		{php $form[\Grido\Components\Operation::ID][\Grido\Components\Operation::ID]->setDisabled()}
		{$form[\Grido\Components\Operation::ID][\Grido\Components\Operation::ID]->control}
		{php $form[\Grido\Grid::BUTTONS][\Grido\Components\Operation::ID]->controlPrototype->class[] = 'hide'}
		{$form[\Grido\Grid::BUTTONS][\Grido\Components\Operation::ID]->control}
	{/if}
</span>

<ul n:if="$paginator->steps && $paginator->pageCount > 1" n:block="paginator" class="pagination" style="display: inline-block; margin: 0 0 -8px 0;">
	<li n:if="$control->page === 1">
		<a n:href="page! page => $paginator->getPage() - 1" aria-label="Previous"><span aria-hidden="true">&laquo;</span></a>
	</li>
	{foreach $paginator->getSteps() as $step}
		<li n:class="$step == $control->page ? active">
			<a href="{$control->link('page!', ['page' => $step])}">{$step}</a>
		</li>
		<li n:if="$iterator->nextValue > $step + 1">
			<a class="prompt" data-grido-prompt="{_'Grido.EnterPage'}" data-grido-link="{$control->link('page!', ['page' => 0])}">...</a>
		</li>
	{/foreach}
	<li n:if="$control->page === $paginator->getPageCount()">
		<a n:href="page! page => $paginator->getPage() + 1" aria-label="Next"><span aria-hidden="true">&raquo;</span></a>
	</li>
</ul>

<span n:block="count" class="count">
	{= sprintf(/*$template->translate('Grido.Items')*/ '%d - %d of %d', $paginator->getCountBegin(), $paginator->getCountEnd(), $control->getCount())}
	{input count}
	{formContainer buttons}
		{input perPage, class => 'hide'}
	{/formContainer}
	<a n:if="$control->hasExport()" class="btn btn-default" href="{=$control->getComponent(\Grido\Components\Export::ID)->link('export!')}" title="{_'Grido.ExportAllItems'}">
		<i class="glyphicon glyphicon-download fa fa-download icon-download"></i>
	</a>
</span>
</div>

{/block}

{/form}
{/snippet}
```

**Comment by matej21** [comment]
do you have the latest version of the plugin? this should be already fixed, see #24

**Comment by hrach** [comment]
today I checked for updates and only other plugings have a new version, but it may be possible that I didn't restarted phpstorm when it was updated few days ago :| :-) so thanks, closing for now, will reopen if it appears again.
